tests/tests/bitasset_tests.cpp

```diff
@@ -127,7 +130,8 @@ BOOST_AUTO_TEST_CASE( reset_backing_asset_on_witness_asset)
       asset_publish_feed_operation op;
       op.publisher = active_witnesses[0];
       op.asset_id = bit_jmj_id;
-      op.feed.settlement_price = op.feed.core_exchange_rate = ~price(bit_usd.amount(1),bit_jmj.amount(300));
+      op.feed.settlement_price = ~price(bit_usd.amount(1),bit_jmj.amount(300));
+      op.feed.core_exchange_rate = ~price(core.amount(1), bit_jmj.amount(300));
```

Comment:
`core` is a reference, which may be invalid here. This affects all following code.

Reply:
I don't doubt what you're saying, but for my own education, what would have invalidated the reference? Could something I or the system have done that would invalidate the reference?

Reply:
`generate_block()` will reevaluate and reapply all pushed transactions, remove and recreate objects, so references to old objects may become invalid.

About the code here, `bit_usd` and `bit_jmj` are local variables with much shorter lifetime, but `core` is not.